agent/engine/common_test.go

```diff
@@ -182,6 +191,45 @@ func validateContainerRunWorkflow(t *testing.T,
 	assertions()
 }
 
+// checkDockerConfigsExceptEnv checks whether the contents in the docker config are expected
+// except for the Env field. Checking for Env field is seperated because when agent converts
+// its container config to docker config, it iterates over the container's env map and
+// append them into docker config's env slice. So the sequence for the env slice is undetermined,
+// and it needs extra logic to check equality.
+func checkDockerConfigsExceptEnv(t *testing.T, expectedConfig *docker.Config, config *docker.Config) {
+	expectedConfigEnvList := expectedConfig.Env
+	configEnvList := config.Env
+	expectedConfig.Env = nil
+	config.Env = nil
+
+	assert.True(t, reflect.DeepEqual(expectedConfig, config),
+		"Mismatch in container config; expected: %v, got: %v", expectedConfig, config)
+
+	expectedConfig.Env = expectedConfigEnvList
+	config.Env = configEnvList
+}
+
+// checkDockerConfigsEnv checks whether two docker configs have same list of environment
+// variables and each has same value.
+func checkDockerConfigsEnv(t *testing.T, expectedConfig *docker.Config, config *docker.Config) {
```

Comment:
Instead of writing this logic, you can use [`assert.ElementsMatch`](https://godoc.org/github.com/stretchr/testify/assert#ElementsMatch) to compare the unordered slice elements.